File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        std::unordered_map<int, int> countMap;
        for(int item : arr) {
            countMap[item]++;
        }
        std::unordered_set<int> values;
        for(const auto i: countMap) {
            if(values.count(i.second)){
                return false;
            } else {
                values.insert(i.second);
            }
        }
        return true;
    }
};

File: 1207-unique-number-of-occurrences/1207-unique-number-of-occurrences.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        arr_map = Counter(arr)
        val_set = set()
        for val in arr_map.values():
            if val not in val_set:
                val_set.add(val)
            else:
                return False
        return True